modules/solid_mechanics/src/materials/ConstitutiveModel.C

```diff
@@ -93,13 +95,19 @@ ConstitutiveModel::applyThermalStrain(unsigned qp,
                                       SymmTensor & strain_increment,
                                       SymmTensor & d_strain_dT)
 {
-  if (_has_temp && _t_step != 0)
+  if (_t_step >= 1)
+    _zeroth_step = false;
+
+  if (_t_step >= 2)
+    _first_step = false;
+
```

Comment:
The above logic looks rather awkward. Rather than two bools why don't you store a copy of `_t_step`?